lib/msf/core/exploit/remote/http/nagios_xi/login.rb

```diff
@@ -0,0 +1,201 @@
+    # This was observed when trying to access Nagios XI 5.3.0 when the license agreement had not been accepted yet.
+    # The `nagiosxi=deleted;` cookie should be filtered out, since it may break authentication
+    if auth_cookies.include?('nagiosxi=deleted;')
+      auth_cookies = auth_cookies.gsub('nagiosxi=deleted;', '').strip
+    end
+
+    # remove duplicate cookies, necessary to make the next check work in case multiple identical cookies were set (as observed on older Nagios versions)
+    auth_cookies_array = auth_cookies.split(' ')
+    auth_cookies_array.uniq!
+    auth_cookies = auth_cookies_array.join(' ')
+
+    # for newer Nagios XI versions, we need to remove the pre_auth cookies from the auth_cookies string, otherwise the session will timeout
+    # however, older Nagios XI versions only use one and the same cookie before and after authentication
+    unless pre_auth_cookies == auth_cookies
+      if auth_cookies.include?(pre_auth_cookies)
+        auth_cookies = auth_cookies.gsub(pre_auth_cookies, '').strip
+      end
+    end
+
+    auth_cookies
+  end
+
+  # performs an HTTP GET request to the Nagios XI backend to verify if authentication succeeded
+  #
+  # @param auth_cookies [String] Cookies required for authentication
+  # @param finish_install [Boolean] Boolean indicating if the module should finish installing Nagios XI on target hosts if the isntallation hasn't been completed or the license agreement is not signed
+  # @return [String, Array] HTTP response body if the Nagios XI backend was accessed successfully, otherwise Array containing an error code and either an error message or (if the license agreement is not signed) an Array with authentication tokens
+  def visit_nagios_dashboard(auth_cookies, finish_intall)
+    # visit the index page to verify we successfully authenticated
+    res_index = send_request_cgi({
+      'method' => 'GET',
+      'uri' => nagios_xi_backend_url,
+      'cookie' => auth_cookies
+    })
+
+    unless res_index
+      return [1, 'Connection failed']
+    end
+
+    unless res_index.code == 200 && res_index.body.include?('>Home Dashboard<')
+      # check if we need to sign the license agreement
+      unless res_index.code == 302 && res_index.headers['Location'].end_with?('login.php?showlicense')
+        return [2, 'Received unexpected reply while trying to acess the NagiosXI home dashboard after authenticating.']
+      end
+
+      print_warning('The Nagios XI license agreement has not yet been signed on the target.')
+      unless finish_intall
+        return [2, 'You can let the module sign the Nagios XI license agreement by setting `FINISH_INSTALL` to true.']
+      end
+
+      nsp = get_nsp(res_index)
+
+      if nsp.blank?
+        return [2, 'Failed to obtain the nsp token required for signing the license agreement.']
+      end
+
+      return [5, [auth_cookies, nsp]]
+    end
+
+    res_index
```

Comment:
Again the format here seems to be `[*code*, *string*]` so might be good to make this conform to that standard.

Reply:
Fix looks good, thanks for addressing this.